xamarin/Xamarin.Forms
# PopModalAsync hides navigation bar on macOS #9536

### Description of Change ###
- added a page in gallery to allow for reproduction of bug
- main change to fix the bug is in Xamarin.Forms.Platform.iOS/DisposeHelpers.cs

This might be problematic due to the fact that part of the dispose code on mac is not run. Someone with a deeper understanding than me should thoroughly take a look. Probably there's a better solution than this that I just couldn't come up with.

### Issues Resolved ### 
- fixes #9536 

### API Changes ###
None

### Platforms Affected ### 
- macOS

### Behavioral/Visual Changes ###
None

### Before/After Screenshots ### 
Not applicable

### Testing Procedure ###
1. Both on iOS (not affected, just test that still everything works) and macOS go to test cases in the gallery app and search for 9536.
2. Select the entry for #9536 - "PopModalAsync...".
3. Tap the button to open modal.
4. Tap the button to close modal.
5. Navigation on top back to search should be there.


### PR Checklist ###
